main.tf

```diff
@@ -90,12 +90,9 @@ resource "argocd_application" "this" {
       namespace = var.namespace
     }
 
+
     sync_policy {
-      automated = {
-        allow_empty = false
-        prune       = true
-        self_heal   = true
-      }
+      automated = var.app_autosync
```

Comment:
I seem to remember that auto-sync is enabled when this "automated" block is present, even if it's empty. Did you test your code ? If it works as expected then it's fine with me.